server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -242,8 +242,16 @@ public Builder dimensions(int dimensions) {
             return this;
         }
 
+        public Builder elementType(ElementType elementType) {
+            this.elementType.setValue(elementType);
+            return this;
+        }
+
         @Override
         public DenseVectorFieldMapper build(MapperBuilderContext context) {
+            // Validate again here because the dimensions or element type could have been set programmatically,
+            // which affects index option validity
+            validate();
```

Comment:
It would be good to make this call non optional and called automatically for all fields before we build the mapper.
Would deserve a separate pr though so just adding it here to not forget.

Reply:
++. Let's make sure to create an issue for this.